### PR TITLE
Feat/rating form route

### PR DESCRIPTION
**Pull request by Scttpr**

### CODE REVIEW

File: models/ratings.mjs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const RatingSchema = new Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+  email: {
+    type: String,
+    required: true,
+  },
+  base: {
```
Comment (Nainterceptor) [inline models/ratings.mjs]: Please use the right naming, here a _base_ is a _campus_.

File: models/ratings.mjs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import createdAtPlugin from './helpers/created-at';
+
+const { Schema } = mongoose;
+
+const RatingSchema = new Schema({
+  name: {
+    type: String,
+    required: true,
+  },
+  email: {
```
Comment (Nainterceptor) [inline models/ratings.mjs]: User will probably don't be logged in. I think it's more relevant to link to the ride with an id than to the user.

File: models/ratings.mjs (hunk 1 above)
Comment (Nainterceptor) [reply models/ratings.mjs]: But, I think we don't want the campus anymore if we link to ride

File: routes/forms.mjs [hunk 3]
```diff
@@ -41,4 +42,34 @@ router.post(
...
+    await Rating.create({
+      name,
+      email,
+      base,
+      uxGrade,
+      recommandationGrade,
+      message,
+    });
+
+    ctx.body = { message: 'Rating sent' };
```
Comment (Nainterceptor) [inline routes/forms.mjs]: I guess you can send a 204

File: models/ratings.mjs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import mongoose from 'mongoose';
+import createdAtPlugin from './helpers/created-at';
+
+const { Schema } = mongoose;
+
+const RatingSchema = new Schema({
+  rideId: {
```
Comment (Nainterceptor) [inline models/ratings.mjs]: Please use the same naming convention that we have everywhere, nest the object to have something like
```
ride: {
  _id: ObjectId
}
```
+ add alias to match _id and id for the 
+ note that type is ObjectId
+ Please ensure that id exists in pre-validation hook

File: routes/forms.mjs [hunk 5]
```diff
@@ -41,4 +42,29 @@ router.post(
   },
 );
 
+router.post(
```
Comment (Nainterceptor) [inline routes/forms.mjs]: Please add unit tests

**Comment by Scttpr** [comment]
Dealt with all requested changes

File: models/status.mjs [hunk 6]
```diff
@@ -122,7 +122,8 @@ export default {
           case DELIVERED:
             await this.sendSMS(
               'Merci d\'avoir fait appel à notre offre de mobilité. '
-              + `Vous pouvez évaluer le service e-Chauffeur : ${this.getSatisfactionQuestionnaireURL()}`,
+              + `Vous pouvez évaluer le service e-Chauffeur :
+              ${this.getSatisfactionQuestionnaireURL()}?rideId=${this.id}}`,
```
Comment (Rodrive) [inline models/status.mjs]: I think we dont need url from config anymore since this is a link to our backend

File: routes/ratings.mjs [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import generateCRUD from '../helpers/abstract-route';
+import Rating from '../models/rating';
+import {
+  CAN_CREATE_RATING,
+  CAN_LIST_RATING,
+} from '../models/rights';
+
+const router = generateCRUD(Rating, {
+  create: {
+    right: [CAN_CREATE_RATING],
```
Comment (Rodrive) [inline routes/ratings.mjs]: If its only one you dont need the array

File: routes/ratings.mjs [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  list: {
+    right: [CAN_LIST_RATING],
```
Comment (Rodrive) [inline routes/ratings.mjs]: Same here

File: routes/ratings.mjs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    async main(ctx) {
+      const { offset, limit } = ctx.parseRangePagination(Rating, { max: 1000 });
```
Comment (Rodrive) [inline routes/ratings.mjs]: Do we have a case where we will need 1000 elements? If not, then default is maybe enough

File: models/role.mjs [hunk 10]
```diff
@@ -30,6 +30,8 @@ export const ROLE_USER = new RoleList(
   rights.CAN_LIST_POI_LOCAL,
   rights.CAN_GET_POI,
   rights.CAN_LIST_POI,
+
+  rights.CAN_CREATE_RATING,
```
Comment (Rodrive) [inline models/role.mjs]: I think an Anonymous user can survey

File: tests/routes/ratings.test.mjs [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+    return [dummyRating, toDropLater];
+  },
+};
+
+describe('Test the rating API endpoint', () => {
+  it(...testCreate(Rating, {
+    ...config,
+    canCall: [generateUserJWTHeader],
+    cannotCall: [generateAnonymousJWTHeader],
```
Comment (Rodrive) [inline tests/routes/ratings.test.mjs]: If an anonymous can survey then this is not true anymore

**Comment by Scttpr** [comment]
[Intégrer questionnaire satisfaction client au site soutenu](https://trello.com/c/U5lrwnIp/130-int%C3%A9grer-questionnaire-satisfaction-client-au-site-soutenu)

**Pull request merged**